### DISCUSSION TITLE
How to visualize it in mujoco?

### DISCUSSION THREAD

**Discussion by JUNLIN1111** [nzolman/sindy-rl#5] [Q&A] [answered]
How to visualize it in mujoco?

**Comment by nzolman** [comment, marked as answer]
Hi @JUNLIN1111—it depends on what you're looking to do here. In terms of visualizing a Mujoco environment from dm_control, I have primarily been using RLlib's dm_control wrapper (using ray 2.6.3) [1] and requesting visual observations from the environment through the `render` function [2]. At a low level, this call's dm_control's `physics.render` function [3].

```python
from ray.rllib.env.wrappers.dm_control_wrapper import DMCEnv

# create env and request 480 x 480 resolution gif.
env = DMCEnv('cartpole', task_name='swingup',height=480, width=480)
env.reset()

# take the null action
obs = env.step(0.0 * env.action_space.sample())

# camera ID can either be 0 or 1, depending on the view you want.
cam_id = 0
pixels = env.render(camera_id=cam_id))
```

Does that meet your needs?

**References**
[1] https://github.com/ray-project/ray/blob/ray-2.6.3/rllib/env/wrappers/dm_control_wrapper.py#L77
[2] https://github.com/ray-project/ray/blob/ray-2.6.3/rllib/env/wrappers/dm_control_wrapper.py#L215
[3] https://github.com/google-deepmind/dm_control/blob/8358f6d605ce239e889851079862accc286c37e1/dm_control/mujoco/engine.py#L178

**Reply by Vcbby** [reply]
Hi there, I am trying to write a script to visualise the cart pole swing task under cart sparse policy, but I am a little bit confused and error occurs when I am coding. Can you show me a simple script to visualise the task pls?

**Comment by JUNLIN1111** [comment]
ber
















At 2025-01-06 02:18:13, "Vcbby" ***@***.***> wrote:

Hi there, I am trying to write a script to visualise the cart pole swing task under cart sparse policy, but I am a little bit confused and error occurs when I am coding. Can you show me a simple script to visualise the task pls?
